docs/contributing/999-rfcs/2024-02-15-resource-pinning.md

```diff
@@ -0,0 +1,185 @@
+---
+title: Resource Pinning
+description: Protection of resources from accidental deletion/moves
+---
+
+# Resource Pinning
+
+- **Author(s):**: @MarkMcCulloh
+- **Submission Date**: 2024-02-15
+- **Stage**: Proposal
+- **Stage Date**: 2024-02-15
+
+Introduce a mechanism to protect resources from accidental deletion/moves in the logical application tree.
+
+## Requirements
+
+### Functional
+- User-ability in sources to signify intent for a resource to not be deleted or moved (e.g. "pinning")
+- Git-trackable way to store changes to pinned resources between compilations, per target platform (e.g. "pinfile")
+- User-ability to manually and declaratively override a pinned resource's path in the resource tree
```

Comment:
I think there's a requirement to be able to control the resource id even if it's not pinned. There could be cases where a library is updated and I want to pin some low level resource within this library to a certain id without having being able to change the library code.